pkg/gatewayserver/io/io.go

```diff
@@ -611,7 +612,6 @@ func (c *Connection) ScheduleDown(path *ttnpb.DownlinkPath, msg *ttnpb.DownlinkM
 		rxErrs = nil
 		if now, ok := c.scheduler.Now(); ok {
 			logger = logger.WithField("now", now)
-			delay = time.Duration(em.Starts() - now)
 		}
```

Comment:
Can we get `now` then also atomically?

Reply:
I'll just return `now` and compute delay inline.